### DISCUSSION TITLE
Can "panda CSS" resolve dynamic props?

### DISCUSSION THREAD

**Discussion by Kazuhiro-Mimaki** [chakra-ui/panda#993] [Q&A] [answered]
Can "panda CSS" resolve dynamic props like following?

```jsx
import { useState } from "react";
import { css } from "styled-system/css";

export default function Index() {
  const [count, setCount] = useState(0);

  return (
    <div className={css({ fontSize: "2xl", fontWeight: "bold" })}>
      Hello 🐼!
      <p className={css({ color: count === 3 ? "red" : "black" })}>
        hello world
      </p>
      <button onClick={() => setCount((prev) => prev + 1)}>plus count</button>
    </div>
  );
}
```

I recognize that "panda CSS" is zero runtime CSS-in-JS framework, so I think above code doesn't work.
However, above code works. 
Is it why? I cannot understand why works regardless of zero runtime.

**Comment by astahmer** [comment, marked as answer]
the extractor is smart enough to extract both condition branches when the condition expression is unresolvable statically, https://panda-css.com/docs/guides/dynamic-styling#runtime-conditions

so, in this example, both "red" and "black" are extracted and will be generated, cause we can't know at build-time which one should be used
https://play.panda-css.com/_Zbe0snK6h
![Screenshot 2023-07-09 at 22 41 50](https://github.com/chakra-ui/panda/assets/47224540/a55e4f28-2fb6-44e2-b9eb-1f98672a4346)
